worldedit-forge/src/main/java/com/sk89q/worldedit/forge/ForgeWorldEdit.java

```diff
@@ -152,56 +154,57 @@ public void onPlayerInteract(PlayerInteractEvent event) {
             return;
         }
 
-        if (!platform.isHookingEvents()) return; // We have to be told to catch these events
+        if (!platform.isHookingEvents())
+            return; // We have to be told to catch these events
 
-        if (event.useItem == Result.DENY || event.entity.worldObj.isRemote) return;
+        if ((event instanceof PlayerInteractEvent.LeftClickBlock
+                && ((PlayerInteractEvent.LeftClickBlock) event)
+                        .getUseItem() == Result.DENY)
+                || (event instanceof PlayerInteractEvent.RightClickBlock
+                        && ((PlayerInteractEvent.RightClickBlock) event)
+                                .getUseItem() == Result.DENY)
+                || event.getEntity().worldObj.isRemote)
+            return;
```

Comment:
Close enough, it's more clear now what's going on at least. Thanks. 

---- On Thu, 05 May 2016 20:10:19 -0700 notifications@github.com wrote ----

In worldedit-forge/src/main/java/com/sk89q/worldedit/forge/ForgeWorldEdit.java:

>  
> -        if (event.useItem == Result.DENY || event.entity.worldObj.isRemote) return;
> +        if ((event instanceof PlayerInteractEvent.LeftClickBlock
> +                && ((PlayerInteractEvent.LeftClickBlock) event)
> +                        .getUseItem() == Result.DENY)
> +                || (event instanceof PlayerInteractEvent.RightClickBlock
> +                        && ((PlayerInteractEvent.RightClickBlock) event)
> +                                .getUseItem() == Result.DENY)
> +                || event.getEntity().worldObj.isRemote)
> +            return;
Fixed, I think.

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub